samples/nextjs/src/components/PersonalizationLoadingComponent.tsx

```diff
@@ -0,0 +1,19 @@
+import { usePersonalization, withComponentFactory } from '@sitecore-jss/sitecore-jss-nextjs';
+import { useI18n } from 'next-localization';
+import { StyleguideComponentProps } from 'lib/component-props';
+
+const PersonalizationLoadingComponent = (props: StyleguideComponentProps): JSX.Element | null => {
+  const { t } = useI18n();
+  const { personalizedComponent, isLoading } = usePersonalization({
+    uid: props.rendering.uid as string,
+    componentFactory: props.componentFactory,
+  });
+
+  return isLoading || personalizedComponent === undefined ? (
```

Comment:
In case if  `personalizedComponent` loaded but undefined - we will have infinite `Loading`?

Reply:
Yes, this is by design. `undefined` means a component is not yet defined, which means "is loading".
The `null` means "nothing to render" - an invisible (hidden) component.
Not null means there is a personalized component to render.

Loading can be infinite only in case the developer made some crucial mistakes in implementation: for instance, somehow managed to use `usePersonalization` hook while the page is not wrapped with personalization HOC.

Reply:
Can it be mentioned somewhere in comments in this case? It's not obvious that `undefined` means that it's loading, and `null` - that it's loaded. Usually we are using boolean variables to identify `loading` state